backend/lib/ycai.js

```diff
@@ -82,16 +82,38 @@ async function fetchRecommendations(videoId, kind) {
   return result;
 }
 
-async function fetchChannelRecommendations(channelId, limit = 20) {
+async function fetchChannelRecommendations(
+  channelId,
+  limit = 20,
+  random = true
+) {
   const mongoc = await mongo3.clientConnect({ concurrency: 1 });
-  const res = await mongo3.readLimit(
-    mongoc,
-    nconf.get('schema').recommendations,
-    { channelId },
-    {},
-    limit,
-    0
-  );
+
+  const aggregate = [
+    {
+      $match: { channelId },
+    },
+  ];
+
+  if (random) {
+    aggregate.push({
+      $sample: { size: limit },
```

Comment:
Is this returning a random chunk of `$limit` length?

Reply:
that's the theory - haven't tested on a db with enough elements though